test.py

```diff
@@ -800,11 +789,21 @@ def __init__(self, test_no: int, shortname: str, suite) -> None:
         self.reject = suite.suite_path / (self.shortname + ".reject")
         self.server_log: Optional[str] = None
         self.server_log_filename: Optional[pathlib.Path] = None
-        CQLApprovalTest._reset(self)
+        self.is_before_test_ok = False
+        self.is_executed_ok = False
+        self.is_new = False
+        self.is_after_test_ok = False
+        self.is_equal_result = False
+        self.summary = "not run"
+        self.unidiff: Optional[str] = None
+        self.server_log = None
+        self.server_log_filename = None
+        self.env: Dict[str, str] = dict()
         self._prepare_args(suite.options)
 
-    def _reset(self) -> None:
+    def reset(self) -> None:
         """Reset the test before a retry, if it is retried as flaky"""
+        super().reset()
```

Comment:
nit, probably we can remove the typing annotations in the `reset()` method. now that we already have them in the constructor?

Reply:
Good catch, removed